cookbook/case_studies/ml_training/house_price_prediction/multiregion_house_price_predictor.py

```diff
@@ -1,19 +1,35 @@
 """
-Predicting House Price in Multiple Regions with XGBoost and Dynamic Workflows
-------------------------------------------------------------------------------
+Predicting House Price in Multiple Regions Using XGBoost and Dynamic Workflows
+-------------------------------------------------------------------------------
+`XGBoost <https://xgboost.readthedocs.io/en/latest/>`__ is an optimized distributed gradient boosting library designed to be efficient, flexible, and portable. 
+It uses `gradient boosting <https://en.wikipedia.org/wiki/Gradient_boosting>`__ technique to implement Machine Learning algorithms.
 
-In this example, you'll use the house price prediction model for one region to expand it to multiple regions. 
+In this tutorial, we will understand how to predict house prices in multiple regions using XGBoost, and dynamic workflows in Flyte.
+
+A dynamic workflow is a typical workflow where the users can perform any arbitrary computation by consuming the inputs and producing the outputs. In the backend, it is designed as a task. During execution, the function body can be run to produce a workflow.
+
+We will split the generated dataset into train, test and validation set. 
+
+Next, we will create two dynamic workflows in Flyte, that will:
+
+1. Generate and split the data for multiple regions.
+
+2. Train the model using XGBoost and generate predictions.
+
+We can use two separate methods to fit the model and generate predictions, but incorporating a workflow will perform both the tasks in parallel, which is efficient and powerful.
```

Comment:
I think this has to be mentioned near the concerning dynamic workflow.

The statement is wrong, though. We don't parallelize the two methods. It has to be:
```suggestion
We can use two different methods to fit the model and generate predictions, but including them in the same dynamic workflow will parallelize the tasks together, where every branch denotes a region.
```
Does this make sense?

Reply:
Asking this for my understanding- if we don't have a dynamic workflow, we create two tasks right? One task would be to fit the model, and other task to make predictions. With dynamic workflow, we are including fit and generate predictions in the same workflow, correct? So, instead of executing two separate tasks, the dynamic workflow ensures that they happen in one pipeline?